src/extensions/agent_modules/shared/mosquitto_utils/src/adu_mosquitto_utils.c

```diff
@@ -140,7 +146,25 @@ bool ADU_are_correlation_ids_matching(
 
     if (outCorrelationData != NULL)
     {
-        if (ADUC_AllocAndStrCopyN(outCorrelationData, responseCorrelationId, (size_t)responseCorrelationIdLen) != 0)
+        // TODO: Remove this workaround once adu service deploys fix that ensures exact correlation id string agent sent is
+        // in the response message. Currently, it parses the UUID without dashes and then introduces dashes in the response
+        // message.
+        char correlation_id_no_dash[37] = "";
+        char* destEnd = NULL;
+        if (!ADUC_CopyIf(
+            correlation_id_no_dash /* dest */,
+            sizeof(correlation_id_no_dash) /* destByteLen */,
+            responseCorrelationId /* src */,
+            (size_t)responseCorrelationIdLen /* srcByteLen */,
+            not_a_dash /* copyPred */,
+            &destEnd /* outDestEnd */))
+        {
+            goto done;
+        }
+
+        byteLenNoDashes = (size_t)(destEnd - correlation_id_no_dash);
+
+        if (!ADUC_AllocAndStrCopyN(outCorrelationData, correlation_id_no_dash, byteLenNoDashes))
```

Comment:
this is horribly confusing to me, and likely going to cause some security issue at some point if not improved.

I see 

* byteLenNoDashes
* outCorrelationDataByteLen
* responseCorrelationIdLen
* n
* sizeof(correlation_id_no_dash)

how is one supposed to track what all of these indicate, and which indicate a buffer size vs a buffersize (minus or including) a null terminator or which match which buffer.

We should look into a better way to do this in C even if means we use the STRING_HANDLE and deal with extra allocations.

Reply:
yeah, I'm removing these out parameters now for security and other reasons.